hashicorp/dev-portal
# refactor: MdxInlineCode

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Moved `code` styles to an MDX `inlineCode` component, and cleans up related styles.

## 🤷 Why

- To avoid unintentional cascading of styles into custom components (such as `code` leaking into `@hashicorp/react-code-block`)
- To consolidate MDX styling in MDX components

## 🛠️ How

- removes `code` styles from `dev-dot-content`, moved to `MdxInlineCode` component
- removes `pre` style with border-radius
    - as far as I can tell, this style wasn't being applied - border radius in `@hashicorp/react-code-block` is applied on a different element, not `pre`
    - intent is to handle further refinements to `@hashicorp/react-code-block` in separate PRs
- removes `pre code` styles that reset aforementioned `code` styles within `@hashicorp/react-code-block`
    - use of `MdxInlineCode` prevents the unintentional cascade, so we no longer need to mitigate it

## 🧪 Testing

- Visit a docs MDX page, such as [/vault/docs/get-started/developer-qs][/vault/docs/get-started/developer-qs], and confirm code blocks and inline code looks as expected (vs [upstream][UPSTREAM/vault/docs/get-started/developer-qs])
- Visit a tutorial MDX page, such as [/waypoint/tutorials/get-started-docker/get-started-ui][/waypoint/tutorials/get-started-docker/get-started-ui], and confirm code blocks and inline code look as expected (vs [upstream][UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui])

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/0/1202072894754377/f
[preview]: https://dev-portal-git-zsconsolidate-mdx-code-css-hashicorp.vercel.app/
[/vault/docs/get-started/developer-qs]: https://dev-portal-git-zsconsolidate-mdx-code-css-hashicorp.vercel.app/vault/docs/get-started/developer-qs
[UPSTREAM/vault/docs/get-started/developer-qs]: https://developer.hashicorp.com/vault/docs/get-started/developer-qs
[/waypoint/tutorials/get-started-docker/get-started-ui]: https://dev-portal-git-zsconsolidate-mdx-code-css-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/get-started-ui
[UPSTREAM/waypoint/tutorials/get-started-docker/get-started-ui]: https://developer.hashicorp.com/waypoint/tutorials/get-started-docker/get-started-ui
